cunumeric/array.py

```diff
@@ -3048,6 +3048,72 @@ def setflags(self, write=None, align=None, uic=None):
         """
         self.__array__().setflags(write=write, align=align, uic=uic)
 
+    @add_boilerplate()
+    def searchsorted(
+        self: ndarray,
+        v: Union[int, float, ndarray],
+        side: str = "left",
+        sorter: Optional[ndarray] = None,
+    ) -> Union[int, ndarray]:
+        """a.searchsorted(v, side='left', sorter=None)
+
+        Find the indices into a sorted array a such that, if the corresponding
+        elements in v were inserted before the indices, the order of a would be
+        preserved.
+
+        Parameters
+        ----------
+        v : scalar or array_like
+            Values to insert into a.
+        side : ``{'left', 'right'}``, optional
+            If 'left', the index of the first suitable location found is given.
+            If 'right', return the last such index. If there is no suitable
+            index, return either 0 or N (where N is the length of a).
+        sorter : 1-D array_like, optional
+            Optional array of integer indices that sort array a into ascending
+            order. They are typically the result of argsort.
+
+        Returns
+        -------
+        indices : int or array_like[int]
+            Array of insertion points with the same shape as v, or an integer
+            if v is a scalar.
+
+        Availability
+        --------
+        Multiple GPUs, Multiple CPUs
+        """
+
+        if self.ndim != 1:
+            raise ValueError("Dimension mismatch: self must be a 1D array")
+
```

Comment:
I think you are missing the case here when self.shape[0]==0:
```
arr = np.array([])
v = np.array([1, 2, 3, 4, 5])
print(arr.searchsorted(v))
```
I think the logic should be:
```
if a.shape[0] == 0:
        return zeros(v.shape, dtype=int64)
```

Reply:
The case with empty self is tested, it works natively because the result-reduction is initialized with `0` or` self.size` (depending on `side`) before the task is started.